odbc/src/handles.rs

```diff
@@ -0,0 +1,202 @@
+use std::{
+    collections::HashSet,
+    sync::{Mutex, RwLock},
+};
+
+use odbc_sys::Integer;
+
+#[derive(Debug)]
+pub struct EnvHandle {
+    pub env: RwLock<Env>,
+}
```

Comment:
question for all the `XHandle` structs:
Do we ever expect these structs to contain anything other than the single `RwLock` field? (I _think_ the answer is no, at least as long as we stick with our whole-handle locking strategy)

If not, I wonder whether it might make the code that uses the handles a bit cleaner to eliminate the need for the intermediate `env` field access by making the `EnvHandle` into either a type alias for `RwLock<Env>` or by making the `env` field private and providing methods directly on `EnvHandle`

These are all half-baked API ergonomics ideas, so just wanted to throw them out there -- happy to discuss further in person, or just feel free to ignore.

Reply:
Let me see how that goes. I like the alias idea.